### PR TITLE
add network rule support in Azure account creation

### PR DESCRIPTION
**Pull request by andyzhangx**
**What type of PR is this?**
/kind feature

**What this PR does / why we need it**:
add network rule support in Azure account creation

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
add network rule support in Azure account creation
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

```docs
none
```

/priority important-soon
/sig cloud-provider
/area provider/azure

### CODE REVIEW

**Comment by andyzhangx** [comment]
@Sakuralbj 
```
#precheck before submit code
hack/verify-golint.sh
hack/verify-gofmt.sh
```

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go [hunk 1]
```diff
@@ -51,18 +52,39 @@ func (az *Cloud) getStorageAccounts(matchingAccountType, matchingAccountKind, re
...
+			if len(accountOptions.VirtualNetworkResourceIDs) > 0 {
+				if acct.AccountProperties == nil || acct.AccountProperties.NetworkRuleSet == nil ||
+					acct.AccountProperties.NetworkRuleSet.VirtualNetworkRules == nil {
+					continue
+				}
+
+				found := false
+				for _, subnetID := range accountOptions.VirtualNetworkResourceIDs {
+					for _, rule := range *acct.AccountProperties.NetworkRuleSet.VirtualNetworkRules {
+						if strings.EqualFold(*rule.VirtualNetworkResourceID, subnetID) && rule.Action == storage.Allow {
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:77]: nit: check nil of `rule.VirtualNetworkResourceID`, or use `to.String(rule.VirtualNetworkResourceID)` replace `*rule.VirtualNetworkResourceID`

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go [hunk 2]
```diff
@@ -119,6 +142,24 @@ func (az *Cloud) EnsureStorageAccount(accountOptions *AccountOptions, genAccount
...
+			// set network rules for storage account
+			var networkRuleSet *storage.NetworkRuleSet
+			virtualNetworkRules := []storage.VirtualNetworkRule{}
+			for _, subnetID := range accountOptions.VirtualNetworkResourceIDs {
+				vnetRule := storage.VirtualNetworkRule{
+					VirtualNetworkResourceID: &subnetID,
+					Action:                   storage.Allow,
+				}
+				virtualNetworkRules = append(virtualNetworkRules, vnetRule)
+				klog.V(4).Infof("subnetID(%s) has been set", subnetID)
```
Comment (feiskyer) [inline staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:154]: nit: `subnetID(%s) has been set for storageAccount %s`

Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go:154]: that's for creating new storage account

File: staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go (hunk 1 above)
Comment (andyzhangx) [reply staging/src/k8s.io/legacy-cloud-providers/azure/azure_storageaccount.go]: good catch, fixed.

**Pull request merged**